tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -126,134 +132,305 @@ void validate(
 }
 } // namespace
 
-TEST_P(DistributedTransformerTest, MLP_Layer) {
-  auto [use_aten_matmul, dtype] = GetParam();
-  std::unique_ptr<Fusion> fusion = std::make_unique<Fusion>();
-  FusionGuard fg(fusion.get());
-  auto mesh = DeviceMesh::createForNumDevices(communicator_->size());
-  at::ScalarType at_dtype = data_type_to_aten(dtype);
+std::vector<at::Tensor> reference_mlp(
+    at::Tensor x,
+    at::Tensor w0,
+    at::Tensor b0,
+    at::Tensor w1,
+    at::Tensor b1,
+    at::ScalarType at_dtype) {
+  at::manual_seed(0);
+  auto linear1 = at::matmul(x, w0).add(b0).to(at::kFloat);
```

Comment:
```suggestion
  auto linear0 = at::matmul(x, w0).add(b0).to(at::kFloat);
```
since you call the params `w0` and `b0`